sway-parse/src/parser.rs

```diff
@@ -21,9 +23,9 @@ impl<'a, 'e> Parser<'a, 'e> {
     }
 
     pub fn emit_error(&mut self, kind: ParseErrorKind) -> ErrorEmitted {
-        let span = match self.token_trees.first() {
-            Some(token_tree) => token_tree.span(),
-            None => {
+        let span = match self.token_trees {
+            [token_tree, ..] => token_tree.span(),
```

Comment:
Using `first`/`Some`/`None` is slightly more robust, eg. because of [this](https://github.com/FuelLabs/sway/pull/2480/commits/52d3bad98d85b3b98338e7645556e0be6fdc23da) (wildcard patterns scare me in general). It's not a blocker on merging this though.